### PR TITLE
add controller

### PR DESCRIPTION
**Pull request by acerlife**

### CODE REVIEW

**Comment by VolodymyrMudryk** [comment]
@acerlife if you do some PR you MUST to add me as reviewer

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/")
+public class CalculatorController {
+    @Autowired
+   private CalculatorDao calculatorDao;
+
+    @GetMapping("/{string}")
+    public @ResponseBody void saveResult(@PathVariable String string) throws Exception{
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: please replace `string` name on `expression`

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        calculatorDao.saveResult(string);
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: you should have the following structure: controller -> service -> dao

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java (hunk 1 above)
Comment (acerlife) [reply src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: is done

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java (hunk 2 above)
Comment (acerlife) [reply src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: is done

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/")
+public class CalculatorController {
+    @Autowired
+   private CalculatorService calculatorService;
+
+    @GetMapping("/{postfix_expression}")
+    public @ResponseBody void saveResult(@PathVariable String postfix_expression) throws Exception{
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: what is it postfix_expression ?

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        calculatorService.saveResult(postfix_expression);
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: where is return?

File: src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.makhnov.springbootcalculator.service;
+
+import com.makhnov.springbootcalculator.dao.CalculatorDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CalculatorService {
+    @Autowired
+    private CalculatorDao calculatorDao;
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java]: CTOR injection

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java (hunk 3 above)
Comment (acerlife) [reply src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: It is postfix expression which we get.

File: src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @Autowired
+    public CalculatorController(CalculatorService calculatorService){
+        this.calculatorService = calculatorService;
+    };
+
+    @GetMapping("/{postfix_expression}")
+    public @ResponseBody
+    HttpStatus saveResult(@PathVariable String postfix_expression) throws Exception{
+       calculatorService.saveResult(postfix_expression);
+       return HttpStatus.OK;
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/controller/CalculatorController.java]: no
I want to see the result on the UI, so you need to return result

File: src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+import com.makhnov.springbootcalculator.dao.CalculatorDao;
+import org.jooq.Field;
+import org.jooq.Record3;
+import org.jooq.SelectJoinStep;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java]: transaction only on save method

File: src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+public class CalculatorService {
+    private final CalculatorDao calculatorDao;
+
+    @Autowired
+    public CalculatorService(CalculatorDao calculatorDao){
+        this.calculatorDao = calculatorDao;
+    }
+
+    public void saveResult(String postfix_expression) throws Exception {
+        calculatorDao.saveResult(postfix_expression);
```
Comment (VolodymyrMudryk) [inline src/main/java/com/makhnov/springbootcalculator/service/CalculatorService.java]: add return

**Pull request merged**